scripts/ninstall.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -euo pipefail
+# GNU coreutil's `install -C` always overrides the destination if the source
+# is not a regular file, which is the case with lots of headers symlinked into
+# the build directory by `./configure`. That causes Rust's Cargo to think that
+# Postgres headers have been updated after `make` call even if no files have been
+# touched. That causes long recompilation of `postgres_ffi` and all dependent
+# packages. To counter that, we handle a special case here: do not copy the file
+# if its content did not change.
+#
+# Such behavior may be incorrect if e.g. permissions have changed, but it should
+# not happen during normal Neon development that often, and rebuild should help.
+#
+# See https://github.com/neondatabase/neon/issues/1873
+if [ "$#" == "5" ]; then
+  if [ "$1" == "-C" ] && [ "$2" == "-m" ] && [ "$3" == "644" ]; then
+    if [ -e "$5" ] && diff -q "$4" "$5" >/dev/null 2>&1; then
+      exit 0
+    fi
+  fi
+fi
+install "$@"
```

Comment:
Took me a while to understand what these if's do.

> if [ "$#" == "5" ]; then

Check that there are exactly 5 arguments. That seemed wrong at first glance, because the Makefile will call install with multiple source files on the same command line. For example:
```
install -C -m 644 /home/heikki/git-sandbox/neon//vendor/postgres-v15/src/include/tsearch/dicts/regis.h /home/heikki/git-sandbox/neon//vendor/postgres-v15/src/include/tsearch/dicts/spell.h /home/heikki/git-sandbox/neon/pg_install/v15/include/postgresql/server/tsearch/dicts
```
However that happens to work for this workaround, as the makefile doesn't use that multi-source variant for the symlinks. Subtle!

Another approach might be to call `readlink` or `realpath` on all the arguments, to resolve the symbolic links, and then call `install` as usual. Not sure if it would be better or faster than this.

Reply:
Exactly so, updated the comment in the file.